tests/integ/modin/index/test_all_any.py

```diff
@@ -0,0 +1,66 @@
+#
+# Copyright (c) 2012-2024 Snowflake Computing Inc. All rights reserved.
+#
+
+import modin.pandas as pd
+import numpy as np
+import pandas as native_pd
+import pytest
+
+import snowflake.snowpark.modin.plugin  # noqa: F401
+from tests.integ.modin.sql_counter import SqlCounter, sql_count_checker
+
+NATIVE_INDEX_BOOL_INT_TEST_DATA = [
+    native_pd.Index([], dtype="object"),
+    native_pd.Index([], dtype=bool),
+    native_pd.Index([], dtype=int, name="NM"),
+    native_pd.Index([True, True, True, True, False, False], dtype=bool, name="name"),
+    native_pd.Index([0, 1, 2, 3, 4], dtype=np.int64),
+    native_pd.Index([True, True, True, None, False, False], dtype=bool),
+]
+
+NATIVE_INDEX_NON_BOOL_INT_TEST_DATA = [
+    native_pd.Index([[1, 2], [2, 3], [3, 4]]),
+    native_pd.Index([3, np.nan, 5], name="my_index"),
+    native_pd.Index([1.1, np.nan]),
+    native_pd.Index([5, None, 7]),
+    native_pd.Index(["a", "b", 1, 2]),
+    native_pd.Index(["a", "b", "c", "d"]),
+    native_pd.Index([5, None, 7]),
+]
+
+
+@pytest.mark.parametrize("native_index", NATIVE_INDEX_BOOL_INT_TEST_DATA)
```

Comment:
I think it would be better to parametrize any/all and just have 2 tests -- one for the negative test and one for the normal test since the code is pretty much replicated. For the negative even the error checking you can do with f-string

Reply:
done!